near-sdk-macros/src/core_impl/abi/mod.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::{hash_map::Entry, HashMap};
+use syn::Type;
+
+pub mod abi_generator;
+pub mod abi_visitor;
+
+pub struct TypeRegistry {
+    pub types: HashMap<Box<Type>, u32>,
```

Comment:
Seems like this should probably be `BTreeMap` since this should always be small. I don't believe it should matter much since this is done at compile time though

Reply:
Yeah I have tried this and the problem is that `Type` does not implement `Ord`